app/dapp/App.js

```diff
@@ -67,20 +67,21 @@ class App extends React.Component {
     const isMainnetConnected =
       mainnet.on && (mainnet.connection.primary.connected || mainnet.connection.secondary.connected)
 
-    const errorComponent =
-      !ready &&
-      (sendDapp.status !== 'ready' && !isMainnetConnected ? (
-        <MainnetDisconnected />
-      ) : sendDapp.status === 'failed' ? (
-        <FailedToLoad />
-      ) : null)
+    const Error = () => {
+      if (!isMainnetConnected) {
+        return <MainnetDisconnected />
+      }
+
+      return <FailedToLoad />
+    }
+    const shouldDisplayError = ready === undefined || sendDapp.status === 'failed'
```

Comment:
this check is weird, it seems to work for now but doesnt really make much sense to check for this variable being `undefined` as it's not clear what that state represents. its a boolean and should only ever be true or false, I think we're just getting away with this because we're not using TS. I think the previous check is more along the lines of what we want: if the send dapp isnt ready for some reason and mainnet isn't connected, we can let the user know. was something about the previous check breaking?

Reply:
Not breaking, nested ternaries 🤮

Reply:
I'm fine w. cleaning those up but just pointing out that the previous check to show the mainnet warning was essentially `!view.ready && sendDapp.status !== 'ready' && mainnetDisconnected` and now its `view.ready !== undefined && mainnetDisconnected` so we lost that middle check

Reply:
Do we want to be displaying an error when the sendDapp status is 'updating'?

Reply:
yeah the statuses are a bit dodgy here. I think its probably still preferable as it probably won't be updating if mainnet isn't connected

Reply:
Ok will test it.

Reply:
Ok updated for this - seems when running dev the send dapp load times are massive, I don't think this is as much of an issue on prod.  However, what if it does take that long on prod?  We're then displaying "Send dapp failed to load" when actually it is technically loading...